lib/salus/npm_audit_wrapper.rb

```diff
@@ -0,0 +1,239 @@
+require 'open3'
+require 'json'
+
+module Salus
+  class NpmAuditWrapper
+    # Structure to hold the relevant information about a given adivsory
+    Advisory = Struct.new(:id, :module, :title, :severity, :url, :prod?, :excepted?)
+
+    # Some common, easily-abbreviated terms to shorten advisory titles
+    TITLE_SUBSTITUTIONS = {
+      'Regular Expression'    => 'Regex',
+      'Denial of Service'     => 'DoS',
+      'Remote Code Execution' => 'RCE'
+    }.freeze
+
+    # Cap title and module name length to avoid the table getting too wide
+    MAX_MODULE_LENGTH = 16
+    MAX_TITLE_LENGTH = 16
+
+    # Headings for the tabulated report
+    TABLE_HEADINGS = ['ID', 'Module', 'Title', 'Sev.', 'URL', 'Prod', 'Ex.'].freeze
+
+    # Terminal colors
+    COLORS = {
+      red: 31,
+      yellow: 33,
+      green: 32
+    }.freeze
+
+    def initialize(stream:, exceptions:, path:, colors: true)
+      @stream = stream
+      @exceptions = exceptions.map(&:to_s).sort_by(&:to_i)
+      @path = path
+      @colors = colors
+    end
+
+    def run!
+      Dir.chdir(@path) do
+        ensure_package_lock! do
+          raw = run_command!('npm audit --json')
+          raw_advisories = JSON.parse(raw, symbolize_names: true).fetch(:advisories).values
+
+          advisories = raw_advisories.map do |radv|
+            # If the advisory exists in a prod dependency, there'll be some finding
+            # where dev is false
```

Comment:
I don't understand what is happening here. Maybe put an example of the hash of one advisory in a comment so it's obvious what this block is doing.